src/interfaces/IPayloadCreator.sol

```diff
@@ -5,7 +5,16 @@ pragma solidity >=0.8.0;
  * @notice Interface for exposing data that oracles can bridge cross-chain.
  */
 interface IPayloadCreator {
+    struct FillRecord {
+        bytes32 orderId;
+        bytes32 outputHash;
+        bytes32 payloadHash;
+    }
+
+    /// @notice Check if a series of fill records are valid.
+    /// @param fills Encoded fill records to validate
+    /// @return bool Whether all fill records are valid
     function arePayloadsValid(
-        bytes32[] calldata payloads
+        bytes calldata fills
     ) external view returns (bool);
```

Comment:
This interface should continue to be generic. The documentation and variable names shouldn't mention fills.

Based on our conversation last week, the only change to the argument should be `bytes32 -> bytes`, i.e. the signature should become `arePayloadsValid(bytes[] calldata payloads)`.